tools/build/src/devfile/meta-yaml-to-devfile-yaml.ts

```diff
@@ -115,6 +115,12 @@ export class MetaYamlToDevfileYaml {
           devfileEndpoint.exposure = 'public';
         }
 
+        // ide type is moved to more generate main endpoint
```

Comment:
```suggestion
        // ide type is moved to more generic main endpoint
```

Reply:
:man_facepalming: thanks!

Reply:
and there: https://github.com/eclipse-che/che-plugin-registry/blob/master/tools/build/src/editor/che-editors-meta-yaml-generator.ts

Reply:
sure, `ide` attributes  are mainly used by the editors (that are specified with che-editors.yaml file)
and then these files are converted to meta.yaml

And we have some other files like che-plugins that are still using a devfile v1 syntax and in that case it's meta.yaml that is translated to devfile.yaml (what you're trying to modify)


but AFAIK we should mainly:
- update che-editors.yaml file to replace `ide` by `main`
- update the converter of devfilev2 syntax to meta.yaml : https://github.com/eclipse-che/che-plugin-registry/blob/master/tools/build/src/editor/che-editors-meta-yaml-generator.ts

Reply:
OK, so you're saying that you would like to see not only devfile moved to main endpoint, but also che-editors.yaml.

let me take a look how I can achieve it.

Reply:
yes because it's where we define the editors (and basically the generated devfile.yaml shouldn't have changes)

Reply:
well, then AFAIU I should:
- change che-editors.yaml to use main type;
- update meta.yaml generator to replace main with ide type;
- change meta.yaml to devfile generator to replace ide type with main type;
:exploding_head:

Reply:
@benoitf Could you tell if Theia relies `type: ide` endpoint attribute?

Reply:
yes Che-Theia grabs the 'ide' type
and to have more fun, che-theia needs to work both with Che-Server and DevWorkspace

Reply:
@benoitf Che Theia has draft to work with main endpoint type, I'll make it ready to review a bit later.
With a newer commit how we should have:
```
- change che-editors.yaml to use main type;
- update meta.yaml generator to replace main with ide type;
- change meta.yaml to devfile generator to replace ide type with main type;
```
and it seems to work as expected:
https://pr-check-963-alpine-che-plugin-registry.surge.sh/v3/plugins/eclipse/che-theia/next/meta.yaml has type ide
https://pr-check-963-alpine-che-plugin-registry.surge.sh/v3/plugins/eclipse/che-theia/next/devfile.yaml has type main